client/src/client.rs

```diff
@@ -22,9 +23,80 @@ use small::SmallStr;
 
 use crate::{
     config::Configuration,
-    http_client::{self, StatusCode, WebSocketError, WebSocketMessage},
+    http::{Headers as HttpHeaders, Method as HttpMethod, RequestBuilder, Response, StatusCode},
+    http_default::{self, DefaultRequestBuilder, WebSocketError, WebSocketMessage},
 };
 
+/// General trait for all response handlers
+pub trait ResponseHandler<O, T = Vec<u8>> {
+    /// Function to parse HTTP response with body `T` to output `O`
+    ///
+    /// # Errors
+    /// May fail by some reason, depends on implementation
+    fn handle(self, response: Response<T>) -> Result<O>;
+}
+
+/// Phantom struct that handles responses of Query API.
+/// Depending on input query struct, transforms a response into appropriate output.
+#[derive(Clone, Copy)]
+pub struct QueryResponseHandler<R>(PhantomData<R>);
+
+impl<R> ResponseHandler<R::Output> for QueryResponseHandler<R>
+where
+    R: Query + Into<QueryBox> + Debug,
+    <R::Output as TryFrom<Value>>::Error: Into<eyre::Error>,
+{
+    fn handle(self, response: Response<Vec<u8>>) -> Result<R::Output> {
+        if response.status() != StatusCode::OK {
+            return Err(eyre!(
+                "Failed to make query request with HTTP status: {}, {}",
```

Comment:
Maybe add `"Response body:"` before the last `{}` so that it will be convinient with `TransactionResponseHandler`?

Reply:
I agree. By the way, I've just copied the previous handling code without any changes.